lib/samples/set_reference_scale/set_reference_scale_sample.dart

```diff
@@ -0,0 +1,206 @@
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        // Add a column to the widget tree.
+        child: Column(
+          children: [
+            Expanded(
+              // Add a map view to the widget tree and set a controller.
+              child: ArcGISMapView(
+                controllerProvider: () => _mapViewController,
+                onMapViewReady: onMapViewReady,
+              ),
+            ),
+            // Add a Settings button to the widget tree.
+            ElevatedButton(
+              // Show the settings dialog when the button is pressed.
+              onPressed: _ready
+                  ? () => showDialog(
+                        context: context,
+                        builder: (context) => showSettings(context),
+                      )
+                  : null,
+              child: const Text('Settings'),
+            )
+          ],
+        ),
+      ),
+    );
+  }
+
+  void onMapViewReady() async {
+    // Create a portal item.
+    final portal = Portal.arcGISOnline();
+    final portalItem = PortalItem.withPortalAndItemId(
+        portal: portal, itemId: '3953413f3bd34e53a42bf70f2937a408');
+    // Load the portal item.
+    await portalItem.load();
+
+    // Create a map from the portal item and load it.
+    _map = ArcGISMap.withItem(portalItem);
+    await _map.load();
+
+    // Get the operational layer names from the map.
+    _allFeatureLayers =
+        _map.operationalLayers.map((layer) => layer.name).toList();
+
+    // Set the map view controller's map to the ArcGIS map.
+    _mapViewController.arcGISMap = _map;
+
+    // Set the ready state variable to true to enable the sample UI.
+    setState(() => _ready = true);
+  }
+
+  StatefulBuilder showSettings(BuildContext context) {
+    // Show the settings dialog.
+    return StatefulBuilder(builder: (context, setNewState) {
+      return Dialog(
+        child: Padding(
+          padding: const EdgeInsets.all(8),
+          child: SingleChildScrollView(
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.spaceAround,
+              children: [
+                Center(
+                  child: Text('Settings',
+                      style: Theme.of(context).textTheme.headlineMedium),
+                ),
+                // Add a dropdown button for setting a new reference scale.
+                DropdownButton(
+                  alignment: Alignment.center,
+                  hint: const Text('Select a New Reference Scale'),
+                  // Set the selected scale
+                  value: _scale,
+                  icon: const Icon(
+                    Icons.arrow_drop_down,
+                    color: Colors.deepPurple,
+                  ),
+                  // Set the onChanged callback to update the selected scale.
+                  onChanged: (newScale) {
+                    setState(() => _scale = newScale!);
+                  },
```

Comment:
I'm currently only seeing the reference scale value updated when a checkbox is selected/unselected, I don't see it update when I use the dropdown. Feels like there is something not quite right with the state update here.